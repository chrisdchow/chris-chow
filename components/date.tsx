import { parseISO, format } from 'date-fns';
import { FunctionComponent } from 'react';

type DateProps = {
  dateString: string;
};

export const Date: FunctionComponent<DateProps> = ({ dateString }) => {
  const date = parseISO(dateString);
  return <time dateTime={dateString}>{format(date, 'LLLL d, yyyy')}</time>;
};
