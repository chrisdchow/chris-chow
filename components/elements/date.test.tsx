import React from 'react';
import { render } from '@test/test-utils';
import { Date } from './date';

describe('Date compnent', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Date dateString='2020-01-01' />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
