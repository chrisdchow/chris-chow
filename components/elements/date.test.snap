// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date compnent matches snapshot 1`] = `
<DocumentFragment>
  <time
    datetime="2020-01-01"
  >
    January 1, 2020
  </time>
</DocumentFragment>
`;
